addons/hr_expense/models/hr_expense.py

```diff
@@ -353,7 +353,7 @@ def create_expense_from_attachments(self, attachment_ids=None, view_type='list')
 
         for attachment in attachments:
             expense = self.env['hr.expense'].create({
-                'name': product.display_name,
+                'name': attachment.name.split('.')[0],
```

Comment:
```suggestion
                'name': os.path.splitext(attachment.name)[1],
```
Do not work if name has '.' in it